src/main/java/games/strategy/util/Match.java

```diff
@@ -172,4 +185,79 @@ public boolean match(final T value) {
       }
     };
   }
+
+  /**
+   * Creates a new match whose condition is satisfied if the test object matches any of the specified conditions.
+   *
+   * @param matches An array of matches; must not be {@code null}.
+   *
+   * @return A new match; never {@code null}.
+   */
+  @SafeVarargs
+  public static <T> Match<T> any(final Match<T>... matches) {
+    checkNotNull(matches);
+
+    return any(Arrays.asList(matches));
```

Comment:
Yeah, I tried that but no joy. :smile:  However, in this case, it only reports the warning once (a comment in the Eclipse bug reasoned that the duplicate warning message is actually a bug in the compiler).  It's possible this warning would go away if `Arrays#stream()` was marked with `@SafeVarargs`, but unfortunately it's not.

The only way I could get rid of both the javac and ECJ warnings was by brute-force converting the array to a `Collection` with a for-each loop.  I didn't want to replace one line of code with five just to get rid of what is apparently an unofficial warning (per the reasoning in the Eclipse bug).

It turns out it's actually pretty simple to shut it off by adding `<< '-Xlint:-varargs'` into our Gradle `compileJava` configuration.  However, it's probably better to have that discussion in a separate issue with everything spelled out so everyone can understand the origin of the `varargs` warnings and debate the pros and cons of turning them off.